app/models/katello/candlepin/content.rb

```diff
@@ -88,6 +95,39 @@ def build_repository
         repository
       end
 
+      def build_docker_repository(docker_container_content_id)
+        registries = product.cdn_resource.get_docker_registries(content.contentUrl)
+        docker_repo = registries.detect do |registry|
+          product.repo_id(content.name, nil, registry['name']) == docker_container_content_id
+        end
+        unless docker_repo
+          Rails.logger.error("Docker registry with pulp_id #{docker_container_content_id} was not found at #{content.contentUrl}")
+          fail _("Docker repository not found")
+        end
+        docker_repo_name = "#{content.name} - (#{docker_repo['name']})"
+        docker_repo_label = ::Katello::Util::Model.labelize(docker_repo_name)
+
+        repository = ::Katello::Repository.new(:environment => product.organization.library,
+                                               :product => product,
+                                               :pulp_id => docker_container_content_id,
+                                               :cp_label => content.label,
+                                               :content_id => content.id,
+                                               :relative_path => relative_path,
+                                               :name => docker_repo_name,
+                                               :label => docker_repo_label,
+                                               :url => docker_repo["url"],
+                                               :feed_ca => ca,
+                                               :feed_cert => product.certificate,
+                                               :feed_key => product.key,
+                                               :content_type => ::Katello::Repository::DOCKER_TYPE,
+                                               :preserve_metadata => true, #preserve repo metadata when importing from cp
+                                               :unprotected => true,
+                                               :content_view_version => product.organization.library.default_content_view_version)
+
+        repository.docker_upstream_name = docker_repo["name"] if repository.docker?
```

Comment:
Seems weird that you're checking `repository.docker?` inside `build_docker_repository`. Are you planning on non-docker repos using this?